addons/gut/test.gd

```diff
@@ -927,6 +927,144 @@ func assert_no_new_orphans(text=''):
 	else:
 		_pass('No new orphans found.' + msg)
 
+# ------------------------------------------------------------------------------
+# Asserts the given setter and getter methods are called when the given property
+# is accessed.
+# ------------------------------------------------------------------------------
+func assert_setget_called(type: Resource, name_property: String
+			, name_setter: String = '', name_getter: String = '') -> void:
+	# arrange
+	var input = {
+			"type": type
+			, "name_property": name_property
+			, "name_setter": name_setter
+			, "name_getter": name_getter
+	}
+	var validation = _validate_assert_setget_called_input(input)
+	if not validation.is_valid:
+		_fail(validation.msg)
+		return
+	
+	var amount_failed_tests = get_fail_count()
+	var obj = _create_obj_from_type(double(type))
+	
```

Comment:
Ok, so the scenario would be, a user already provides a doubled instance, no? Maybe i can add a test case for this scenario as well.

Reply:
No, that wasn't the case but it looks like you fixed it or I missed something in my original post.  Here's the scenario I was thinking:

```
var Foo = load('res://foo.gd')

func before_each():
    # This line stubs ALL doubles of Foo to call super's get_foo unless stub is 
    # called on an instance.
    stub(Foo, 'set_bar').to_call_super()

func test_bar_property():
    assert_setget_called(Foo, 'foo', 'set_bar', 'get_bar')
```

If I remember correctly, you were doing a `doubule` of Foo.  In the scenario above, without explicitly stubbing `set_bar` then the super's `set_bar` would have been called and the `null` you are passing could cause an error depending on the implementation of `set_bar`.  There are  also scenarios where errors could pop up for `get_bar` as well.

Reply:
For the setter, it might be better to do:
```
stub(obj, name_setter).when_passed(null).to_do_nothing()
```
I don't think that there could be any issues since you are stubbing an instance that you have created, but just to be  sure and to be explicit this might be better.

Reply:
I see. Now I understand better. Thanks for addressing it. I will change the code as suggested.

Reply:
As far as I can see, this is the last open point.

I looked into the code and right now, we have the following
```
	var expected_calls_getter = 0
	var obj = _create_obj_from_type(double(type))

	if name_setter != '':
		expected_calls_setter = 1
		stub(obj, name_setter).to_do_nothing()
		obj.set(name_property, null)
		amount_calls_setter = gut.get_spy().call_count(obj, str(name_setter))
```
I'm a little bit lost on the suggested changes after looking into it again. Why would it be better to exchange `stub(obj, name_setter).to_do_nothing()` with `stub(obj, name_setter).when_passed(null).to_do_nothing()`?

My idea was to always stub it to do nothing since we technically just need to know whether the method was called. We do not care what it actually does. 

If you prefer to change it I can do, no problem.